compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -3211,17 +3211,18 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
       cls.is(Sealed) && !cls.hasAnonymousChild
 
     def decompose(cls: Symbol): List[Symbol] =
-      cls.children.flatMap { child =>
+      cls.children.map: child =>
         if child.isTerm then
-          child.info.classSymbols // allow enum vals to be decomposed to their enum class (then filtered out) and any mixins
-        else child :: Nil
-      }.filter(child => child.exists && child != cls)
+          child.info.classSymbol.orElse(child)
+        else child
+      .filter(child => child.exists && child != cls)
```

Comment:
The change looks good to me. However it means that recursive calls to `provablyDisjointClasses` provide arguments that are *not* classes! We should make that clear in a the doc comment of `provablyDisjointClasses`.

Also we should adjust the spec test at
https://github.com/scala/scala3/blob/590691b09bb8554dd312a4e942f91dcd43ba6b44/docs/_spec/03-types.md?plain=1#L1074-L1080
to account for that possibility. Perhaps
```diff
-Second, we define the relation ´⋔´ on *classes* (including traits and hidden classes of objects) as:
+Second, we define the relation ´⋔´ on *classes* (including traits and hidden classes of objects and enum values) as:
```